logic/concludable/ConjunctionConcludable.java

```diff
@@ -112,21 +119,86 @@ public Value asValue() {
-
-            // Find all roleplayer mapping combinations, which should have the form:
-            // Set<List<Pair<RelationConstraint.RolePlayer, RelationConstraint.RolePlayer>>> rolePlayerMappings
-            // For each, prune if there is no label intersection (or any other pruning, e.g. by value)
-            // Then build a Unification for each valid combination
-            return Stream.empty(); // TODO
+        public Stream<Map<Reference, Set<Reference>>> unify(HeadConcludable.Relation unifyWith) {
+            if (this.constraint().players().size() > unifyWith.constraint().players().size()) return Stream.empty();
+            Map<Reference, Set<Reference>> startingUnifier = new HashMap<>();
+            Optional<Map<Reference, Set<Reference>>> newUnifier;
+            if (constraint.owner().reference().isName()) {
+                newUnifier = updatedUnifier(this.constraint().owner(), unifyWith.constraint().owner(), startingUnifier);
+                if (newUnifier.isPresent()) startingUnifier = newUnifier.get();
+                else return Stream.empty();
+            }
+
+            if (constraint.owner().isa().isPresent() && constraint.owner().isa().get().type().reference().isName()) {
+                assert unifyWith.constraint().owner().isa().isPresent();
+                newUnifier = updatedUnifier(this.constraint().owner().isa().get().type(),
+                        unifyWith.constraint().owner().isa().get().type(), startingUnifier);
+                if (newUnifier.isPresent()) startingUnifier = newUnifier.get();
+                else return Stream.empty();
+            }
+
+            return matchRolesAndUnify(new ArrayList<>(this.constraint().players()), unifyWith.constraint().players(),
+                    startingUnifier, new HashSet<>()).stream();
+        }
+
+        private Set<Map<Reference, Set<Reference>>> matchRolesAndUnify(
+                List<RolePlayer> conjunctionRolePlayers, List<RolePlayer> headRolePlayers,
+                Map<Reference, Set<Reference>> unifier, Set<RolePlayer> visitedHeadRolePlayers) {
+            Set<Map<Reference, Set<Reference>>> allUnifiers = new HashSet<>();
+            if (conjunctionRolePlayers.isEmpty()) {
+                return set(unifier);
+            }
+            RolePlayer conj = conjunctionRolePlayers.remove(0);
+            for (RolePlayer head : headRolePlayers) {
+                if (visitedHeadRolePlayers.contains(head)) continue;
+                visitedHeadRolePlayers.add(head);
+                updatedUnifier(conj, head, unifier).ifPresent(newUnifier ->
+                        allUnifiers.addAll(matchRolesAndUnify(conjunctionRolePlayers,
+                                headRolePlayers, newUnifier, visitedHeadRolePlayers)));
+                visitedHeadRolePlayers.remove(head);
+            }
+            conjunctionRolePlayers.add(conj);
+            return allUnifiers;
+        }
+
+        Optional<Map<Reference, Set<Reference>>> updatedUnifier(
```

Comment:
don't use past tense in general - should be `extendUnifier` or something

Reply:
OK, changed to `extendUnifier`.